extensions/amp-consent/0.1/consent-ui.js

```diff
@@ -666,10 +699,14 @@ export class ConsentUI {
     }
     setImportantStyles(this.parent_, {
       transform: `translate3d(0px, calc(100% - ${this.initialHeight_}), 0px)`,
+      '--lightbox-height': `${this.initialHeight_}`,
     });
-    if (this.enableBorder_) {
-      const {classList} = this.parent_;
+    const {classList} = this.parent_;
+    if (this.enableBorder_ && !this.lightboxEnabled_) {
       classList.add(consentUiClasses.enableBorder);
+    } else if (this.lightboxEnabled_) {
+      classList.add(consentUiClasses.enableBorderLightbox);
+      this.sendViewerMessage_(REQUEST_OVERLAY);
```

Comment:
Flip to simplify conditional clause.
```suggestion
    if (this.lightboxEnabled_) {
      classList.add(consentUiClasses.enableBorderLightbox);
      this.sendViewerMessage_(REQUEST_OVERLAY);
    } else if (this.enableBorder_) {
      classList.add(consentUiClasses.enableBorder);
```

Reply:
1. Also, consider flipping `enableBorder_` to match `lightboxEnabled_`, that way you can use a compound selector and only send the message for lightbox.

2. In any case, you should use the `Viewport` service for this, don't talk to the viewer directly.


```suggestion
    if (this.enableBorder_) {
      // This classname is applied differently by CSS selector
      classList.add(consentUiClasses.enableBorder);
    } 
    if (this.lightboxEnabled_) {
      this.viewport_.enterLightboxMode();
    }
```

Reply:
> In any case, you should use the Viewport service for this, don't talk to the viewer directly.

Following the precedence set here #23491; basically `this.viewport_.enterLightboxMode()` does more work than we need.

Reply:
Which work is unnecessary? If it doesn't affect the use case negatively, I'd say it'd be better to consolidate implementations rather than duplicate them.

Reply:
Yeah, I agree.